Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -3200,6 +3231,9 @@ static Color parseColorFunctionRaw(CSSParserTokenRange& range, const CSSParserCo
     case CSSValueColorMix:
         color = parseColorMixFunctionParametersRaw(colorRange, context);
         break;
+    case CSSValueLightDark:
+        // FIXME: Need a worker-safe way to compute light-dark colors.
```

Comment:
Can't we just have something that exclusively uses the CSS color-scheme CSS property on the same renderstyle? Workers don't have a document so the meta tag isn't necessary to check there.

Reply:
This method needs to return a resolved `Color`, and the only entry point to this logic is here:

```
#if ENABLE(OFFSCREEN_CANVAS)
    if (is<OffscreenCanvas>(canvasBase))
        return CSSPropertyParserWorkerSafe::parseColor(colorString);
#endif
```

I'm not sure we even have a `RenderStyle` for `OffscreenCanvas`? And if there's no `Document` or `Page`, we also won't be able to get the system color scheme preference.

Though maybe this means we should always resolve to the light color?

Reply:
There are 2 possible options:
- always returning the light color
- returning a color based on the canvas' color scheme, sort of how currentColor is computed: https://searchfox.org/wubkat/rev/2cb2f84b68df8a519e252d535de43e7d19567883/Source/WebCore/html/canvas/CanvasStyle.cpp#80-93

Reply:
I think the second option is preferable if a backing canvas is available, see:

https://github.com/whatwg/html/issues/9940
https://drafts.csswg.org/css-color-4/#parse-color

Reply:
Well, in workers there's no element. So I think that would have to be explicitly defined first.